mock-hpos-api/index.js

```diff
@@ -0,0 +1,147 @@
+const express = require('express')
+const cors = require('cors')
+const bodyParser = require('body-parser')
+const stringify = require('fast-json-stable-stringify')
+const _ = require('lodash')
+const HpAdminKeypair = require('@holo-host/hp-admin-keypair').HpAdminKeypair
+const { signPayload, hashString } = require('./authUtils')
+const defaultResponse = require('./defaultResponse')
+
+const HC_PUBKEY = '5m5srup6m3b2iilrsqmxu6ydp8p8cr0rdbh4wamupk3s4sxqr5'
+
+const API_ROOT = '/api/v1'
+
+function removeApiRoot (path) {
+  if (path.indexOf(API_ROOT) == 0) {
+    return path.slice(API_ROOT.length);
+  } else {
+    return path
+  }
+}
+
+function generateResponseKey (method, path, data) {
+  return `${method}:${removeApiRoot(path)}:${JSON.stringify(data)}`
+}
+
+// NEXT_PATH is used to key a next response in place of an actual path. It's required to be different from any value that could come from req.path
+const NEXT_PATH = ' . next . '
+
+const NEXT_RESPONSE_KEY = generateResponseKey('', NEXT_PATH, {})
+
+class MockHposApi {
+  constructor(port, authEmail, authPassword) {
+    this.port = port
+    this.shouldCheckAuth = !!authEmail && !!authPassword
+    this.authEmail = authEmail
+    this.authPassword = authPassword
+    
+    this.initializeResponses()
+  }
+
+  static async start (port, authEmail, authPassword) {
+    const mockHposApi = new MockHposApi(port, authEmail, authPassword)
+
+    const app = express()
+
+    app.use(cors())
+    app.use(bodyParser.json())
+    app.use(mockHposApi.checkAuth.bind(mockHposApi))
+    app.use(mockHposApi.handleRequest.bind(mockHposApi))
+
+    await mockHposApi.startServer(app)
+
+    return mockHposApi
+  }
+
+  // this is more convoluted than I'd like because I want init to be a blocking call, so have to uncallback app.listen
+  startServer (app) {
+    return new Promise((resolve, reject) => {
+      this.server = app.listen(this.port, () => {
+        console.log(`Mock hpos-admin-api listening at http://localhost:${this.port}`)
+        resolve()
+      }).on('error', reject)
+    })
+  }
+
+  async checkAuth (req, res, next) {
+    if (!this.shouldCheckAuth) {
+      next()
+    } else {
+
+      const { method, path, body } = req
+
+      const keypairInstance = new HpAdminKeypair(HC_PUBKEY, this.authEmail, this.authPassword)
+
+      const signature = await signPayload(keypairInstance, method, path, body)
+
+      if (signature === req.header('x-hpos-admin-signature')) {
+        next()
+      } else {
+        res.status(401).end()
+      }
+    }
+  }
+
+  any (response) {
+    this.anyResponse = response
+  }
```

Comment:
What is the purpose of this function? From a quick ctrl-f it looks like it's never called

Reply:
`.any`, `.next` and `.once` are not currently used by any tests. But they're there for use in anticipation of future tests. The structure of this mock comes from the mock-conductor, so it mostly didn't require writing a bunch of extra code, so I decided to leave them in for the future. But this is breaking the principle of [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) and it would be reasonable to remove these and only add them once they're needed